Odata-docs/webapi-8/fundamentals/client-driven-paging.md

```diff
@@ -0,0 +1,173 @@
+---
+title:  "Client-driven paging in ASP.NET Core OData 8"
+description: Client-driven paging in ASP.NET Core OData 8.
+date:   2023-01-10
+ms.date: 1/10/2023
+author: habbes
+ms.author: clhabins
+---
+
+# Client-driven paging in ASP.NET Core OData 8
+**Applies To**:[!INCLUDE[appliesto-webapi](../../includes/appliesto-webapi-v8.md)]
+
+The client can implement paging by telling the OData service how many items to return for a given request. The `$top` query option is used to limit the number of results and the `skip` query option used to specify the offset. For example `GET /Customers?$skip=3$top=5` will return up to 5 items starting from the 4th item in the collection (i.e. after skipping the first 3).
+
+Using these two query options the client can divide a large collection into separate pages and fetch the pages in separate requests. For example, if our collection has 25 items, and we want each page to have 10 items, then we can fetch the different page as follows:
+- First page: `$top=10`
+- Second page: `$skip=10&$top=10`
+- Third page: `$skip=20&top=10` (The last page will return only 5 items)
+
+By default, ASP.NET Core OData 8 limits the maximum value for `$top` to 0. So if you add `$top=2` to your request, you would get an error with a message like:
+
+```json
+{
+    "error": {
+        "code": "",
+        "message": "The query specified in the URI is not valid. The limit of '0' for Top query has been exceeded. The value from the incoming request is '2'."
+}
+```
+
+You can configure the maximum top value using the [`ODataOptions.SetMaxTop()`](/dotnet/api/microsoft.aspnetcore.odata.odataoptions.setmaxtop) when adding OData services to your application. The following snippet would set the maximum allowed `$top` value to 100.
+
+```c#
```

Comment:
I just noticed that \`\`\`c# renders a little different from \`\`\`csharp

\`\`\`c#:
![image](https://user-images.githubusercontent.com/28291332/214649759-14704bd6-383f-46ff-bee9-09b29b85c059.png)

vs

\`\`\`csharp:
![image](https://user-images.githubusercontent.com/28291332/214649937-23bc7a80-cf19-41e0-9d1b-5fae6e181091.png)

Because of this, I think I'd suggest that you use \`\`\`csharp given that even \`\`\`json is rendered as:
![image](https://user-images.githubusercontent.com/28291332/214650320-d20a82d9-3c26-4204-8f63-001bfe7f8763.png)

Reply:
You mean the language on the title is uppercase (C#) when using `csharp` and lowercase (c#) when using `c#`? I never noticed that before. Thanks.